pkg/nuctl/command/export.go

```diff
@@ -0,0 +1,275 @@
+	cmd            *cobra.Command
+	rootCommandeer *RootCommandeer
+}
+
+func newExportCommandeer(rootCommandeer *RootCommandeer) *exportCommandeer {
+	commandeer := &exportCommandeer{
+		rootCommandeer: rootCommandeer,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "export",
+		Short: "Export resource to json format",
+	}
+
+	exportFunctionCommand := newExportFunctionCommandeer(commandeer).cmd
+	exportProjectCommand := newExportProjectCommandeer(commandeer).cmd
+
+	cmd.AddCommand(
+		exportFunctionCommand,
+		exportProjectCommand,
+	)
+
+	commandeer.cmd = cmd
+
+	return commandeer
+}
+
+type exportFunctionCommandeer struct {
+	*exportCommandeer
+	getFunctionsOptions platform.GetFunctionsOptions
+	output              string
+	noScrub             bool
+}
+
+func newExportFunctionCommandeer(exportCommandeer *exportCommandeer) *exportFunctionCommandeer {
+	commandeer := &exportFunctionCommandeer{
+		exportCommandeer: exportCommandeer,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "function [name[:version]]",
+		Aliases: []string{"fu"},
+		Short:   "Export function to yaml format",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// if we got positional arguments
+			if len(args) != 0 {
+
+				// second argument is a resource name
+				commandeer.getFunctionsOptions.Name = args[0]
+			}
+
+			// initialize root
+			if err := exportCommandeer.rootCommandeer.initialize(); err != nil {
+				return errors.Wrap(err, "Failed to initialize root")
+			}
+
+			commandeer.getFunctionsOptions.Namespace = exportCommandeer.rootCommandeer.namespace
+
+			functions, err := exportCommandeer.rootCommandeer.platform.GetFunctions(&commandeer.getFunctionsOptions)
+			if err != nil {
+				return errors.Wrap(err, "Failed to get functions")
+			}
+
+			if len(functions) == 0 {
+				if commandeer.getFunctionsOptions.Name != "" {
+					return nuclio.NewErrNotFound("No functions found")
+				}
+				cmd.OutOrStdout().Write([]byte("No functions found")) // nolint: errcheck
+				return nil
+			}
+
+			// render the functions
+			return renderFunctions(functions, commandeer.output, cmd.OutOrStdout(), commandeer.renderFunctionConfig)
+		},
+	}
+
+	cmd.PersistentFlags().StringVarP(&commandeer.getFunctionsOptions.Labels, "labels", "l", "", "Function labels (lbl1=val1[,lbl2=val2,...])")
+	cmd.PersistentFlags().StringVarP(&commandeer.output, "output", "o", OutputFormatYAML, "Output format - \"yaml\", or \"json\"")
+	cmd.PersistentFlags().BoolVar(&commandeer.noScrub, "no-scrub", false, "Allow function sensitive data to be exported")
+
+	commandeer.cmd = cmd
+
+	return commandeer
+}
+
+func (e *exportFunctionCommandeer) renderFunctionConfig(functions []platform.Function, renderer func(interface{}) error) error {
+	if len(functions) == 1 {
+		functionConfig := functions[0].GetConfig()
+		functionConfig.PrepareFunctionForExport(e.noScrub)
+		if err := renderer(functionConfig); err != nil {
+			return errors.Wrap(err, "Failed to render function config")
+		}
+		return nil
+	}
```

Comment:
code dup? why the special case of `len(functions) == 1`

Reply:
when a single function is exported, output looks like:
```
meta:
   # stuff
spec:
   # stuff
```

if multiple functions are exported:
```
function1:
   meta:
      # stuff
   spec:
      # stuff
function2:
   meta:
      # stuff
   spec:
      # stuff
```

Not duplication